examples/notebooks/plot_grid_search_census.py

```diff
@@ -0,0 +1,216 @@
+
+from fairlearn.widget import FairlearnDashboard
+from sklearn.model_selection import train_test_split
+from fairlearn.reductions import GridSearch
+from fairlearn.reductions import DemographicParity, ErrorRate
+
+from sklearn.preprocessing import LabelEncoder, StandardScaler
+from sklearn.linear_model import LogisticRegression
+import pandas as pd
+import shap
+
+# %%
+# We can now load and inspect the data from the :code:`shap` package:
+
+X_raw, Y = shap.datasets.adult()
+X_raw
+
+# %%
+# We are going to treat the sex of each individual as a protected
+# attribute (where 0 indicates female and 1 indicates male), and in
+# this particular case we are going separate this attribute out and drop it
+# from the main data.
+# We then perform some standard data preprocessing steps to convert the
+# data into a format suitable for the ML algorithms
+
+A = X_raw["Sex"]
+X = X_raw.drop(labels=['Sex'], axis=1)
+X = pd.get_dummies(X)
+
+sc = StandardScaler()
+X_scaled = sc.fit_transform(X)
+X_scaled = pd.DataFrame(X_scaled, columns=X.columns)
+
+le = LabelEncoder()
+Y = le.fit_transform(Y)
+
+# %%
+# Finally, we split the data into training and test sets:
+
+X_train, X_test, Y_train, Y_test, A_train, A_test = train_test_split(X_scaled,
+                                                                     Y,
+                                                                     A,
+                                                                     test_size=0.2,
+                                                                     random_state=0,
+                                                                     stratify=Y)
+
+# Work around indexing bug
+X_train = X_train.reset_index(drop=True)
+A_train = A_train.reset_index(drop=True)
+X_test = X_test.reset_index(drop=True)
+A_test = A_test.reset_index(drop=True)
+
+# Improve labels
+A_test = A_test.map({0: "female", 1: "male"})
+
+# %%
+# Training a fairness-unaware predictor
+# -------------------------------------
+#
+# To show the effect of Fairlearn we will first train a standard ML predictor
+# that does not incorporate fairness.
+# For speed of demonstration, we use the simple
+# :class:`sklearn.linear_models.LogisticRegression` class:
+
+unmitigated_predictor = LogisticRegression(solver='liblinear', fit_intercept=True)
+
+unmitigated_predictor.fit(X_train, Y_train)
+
+# %%
+# We can load this predictor into the Fairness dashboard, and examine how it
+# is unfair (there is a warning about AzureML since we are not yet integrated
+# with that product):
+
+FairlearnDashboard(sensitive_features=A_test, sensitive_feature_names=['sex'],
+                   y_true=Y_test,
+                   y_pred={"unmitigated": unmitigated_predictor.predict(X_test)})
+
+# %%
+# Looking at the disparity in accuracy, we see that males have an error
+# about three times greater than the females.
+# More interesting is the disparity in opportunitiy - males are offered loans at
```

Comment:
men/women instead of males/females?

Reply:
Why? I can change it throughout to be consistent, but what's the motivation for doing so?